.github/workflows/equivalence-test.yml

```diff
@@ -1,27 +1,238 @@
+
+      - name: "Upload terraform-equivalence-testing binary"
+        uses: actions/upload-artifact@v2
+        with:
+          name: terraform-equivalence-testing
+          path: terraform-equivalence-testing
+
+  prepare-terraform:
+    name: "Prepare Terraform binary"
+    if: ${{ needs.get-metadata.outputs.run-equivalence-tests == 'true' }}
+    runs-on: ubuntu-latest
+    needs:
+      - get-metadata
+
+    env:
+      terraform-version: ${{ needs.get-metadata.outputs.terraform-version }}
+      build-run-id: ${{ needs.get-metadata.outputs.build-run-id }}
+
+    steps:
+      - name: "Download terraform binary"
+        run: |
+          curl \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.ELEVATED_GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/hashicorp/terraform/actions/runs/${{ env.build-run-id }}/artifacts" > artifacts.json
+          
+          VERSION="${{ env.terraform-version }}" # The Terraform artifacts don't have the `v` prefix.
+          ARTIFACT="terraform_${VERSION#v}_${{ env.target-os }}_${{ env.target-arch }}.zip"
+          ARTIFACT_ID=$(jq -r --arg ARTIFACT "$ARTIFACT" '.artifacts | .[] | select(.name == $ARTIFACT) | .id' artifacts.json)
+          
+          curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.ELEVATED_GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/hashicorp/terraform/actions/artifacts/$ARTIFACT_ID/zip" > "$ARTIFACT.zip"
+
+      - name: "Unzip terraform binary"
+        run: |
+          VERSION="${{ env.terraform-version }}" # The Terraform artifacts don't have the `v` prefix.
+          ARTIFACT="terraform_${VERSION#v}_${{ env.target-os }}_${{ env.target-arch }}.zip"
+          
+          # We actually have nested zip files, as the Github API downloads the
+          # artifacts in a zip file and the Terraform build action embeds the
+          # terraform binary in a zip file also.
+          
+          unzip $ARTIFACT.zip
+          unzip $ARTIFACT
+
+      - name: "Upload terraform binary"
+        uses: actions/upload-artifact@v2
+        with:
+          name: terraform
+          path: terraform
+
+  run-equivalence-tests:
+    name: "Run equivalence tests"
+    if: ${{ needs.get-metadata.outputs.run-equivalence-tests == 'true' }}
+    runs-on: ubuntu-latest
+    needs:
+      - get-metadata
+      - prepare-terraform
+      - prepare-equivalence-tests
+
+    env:
+      target-branch: ${{ needs.get-metadata.outputs.target-branch }}
+      terraform-version: ${{ needs.get-metadata.outputs.terraform-version }}
+
+    steps:
+      - name: "Checkout repository at target branch ${{ env.target-branch }}"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.target-branch }}
+          token: ${{ secrets.ELEVATED_GITHUB_TOKEN }} # Elevated token as we intend to push updates back.
```

Comment:
Same thought here as for my comment on the other use of `ELEVATED_GITHUB_TOKEN` above. Can we specify `contents: write` permission just for this job and thus avoid the need for an explicitly-issued token?

Reply:
This is done!